src/rust/ide/view/graph-editor/src/component/visualization/container/visualization_chooser.rs

```diff
@@ -100,23 +106,31 @@ impl VisualizationChooser {
 
             // === Input Processing ===
 
-            eval frp.input.set_entries ([model,menu](alternatives) {
-                model.visualization_alternatives.replace(alternatives.clone());
-                let alternatives:list_view::entry::AnyModelProvider  = alternatives.clone().into();
-                menu.set_entries.emit(alternatives);
-            });
+            eval  frp.set_icon_size ((size) menu.set_icon_size.emit(size) );
+            eval  frp.set_icon_padding ((size) menu.set_icon_padding.emit(size) );
+            eval_ frp.hide_selection_menu ( menu.hide_selection_menu.emit(()) );
+            eval  frp.set_menu_offset_y ((offset) menu.set_menu_offset_y.emit(offset) );
+
+            _set_selected <- frp.input.set_selected.map2(&frp.output.entries,
+                f!([menu](selected,entries) {
+                    if let Some(selected) = selected {
+                        let ix = entries.iter().position(|item| item == selected);
+                        menu.set_selected.emit(ix);
+                    }
+                })
+            );
```

Comment:
Could you write it more functional than imperative? I mean something along the lines of:

```rust
            ix <= frp.input.set_selected.map2(&frp.output.entries,
                f!([menu](selected,entries) selected.map(|s| entries.iter().position(|item| item == selected)))
            );
            eval ix ((ix) menu.set_selected.emit(ix));
```

Btw the last line should be `menu.set_selected <+ ix` in FRP3.

The rule of thumb for me is that if we can "emit" things as the ONLY operation in eval / map, we should do it this way.